### PR TITLE
Removes ios universal link vmservices and let xcodeproject to dump js…

### PR DESCRIPTION
**Pull request by chunhtai**
…on file

The deeplink validation tool will become an static app so it can't no longer access vm services.

The goal will be then to turn them into flutter analyze command similar to `flutter analyze --android --[options]` that static app can use on.

This pr only removes vm services and turn the api to dump a output file instead of printing everything to stdout.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by vashworth — COMMENTED** [review]
It seems that your new method `outputUniversalLinkSettings` is not called anywhere (other than tests). Is that intentional?

File: packages/flutter_tools/lib/src/xcode_project.dart [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:convert';
```
Comment (vashworth) [inline packages/flutter_tools/lib/src/xcode_project.dart:5]: Error from failing tests:
```
lib/src/xcode_project.dart imports 'dart:convert'; import 'lib/src/convert.dart' instead
```

**Comment by chunhtai** [comment]
> It seems that your new method `outputUniversalLinkSettings` is not called anywhere (other than tests). Is that intentional?

There will be another pr that wire up outputUniversalLinkSettings to flutter analyze command

**Review by vashworth — CHANGES_REQUESTED** [review]
Looks like you also need to update `integration.shard/vmservice_integration_test.dart`

**Review by vashworth — APPROVED** [review]
LGTM

**Pull request merged**